core-bundle/src/Resources/contao/pages/PageLogout.php

```diff
@@ -31,19 +31,14 @@ class PageLogout extends Frontend
 	 */
 	public function getResponse($objPage)
 	{
-		// Set last page visited
-		if ($objPage->redirectBack)
-		{
-			$_SESSION['LAST_PAGE_VISITED'] = $this->getReferer();
-		}
-
 		$strLogoutUrl = System::getContainer()->get('security.logout_url_generator')->getLogoutUrl();
 		$strRedirect = Environment::get('base');
+		$strReferer = $this->getReferer();
 
 		// Redirect to last page visited
-		if ($objPage->redirectBack && !empty($_SESSION['LAST_PAGE_VISITED']))
+		if ($objPage->redirectBack && !empty($strReferer))
```

Comment:
```suggestion
		if ($this->redirectBack && ($strReferer = $this->getReferer()))
```

Reply:
Changed in 0c914d124acf5c0ec87df3721b74e1cdfeebd4d7